src/lib/utils.h

```diff
@@ -56,6 +59,24 @@
         }                                                      \
     } while (/* CONSTCOND */ 0)
 
+/*
+ * @params
+ * array:       array of the structures to lookup
+ * id_field     name of the field to compare against
+ * ret_field    filed to return
+ * id           id to look for
+ * ret          return value
+ */
+#define LOOKUP_ARRAY_BY_ID(array, id_field, ret_field, id, ret) \
+    do {                                                        \
+        for (size_t __i = 0; __i < ARRAY_SIZE(array); __i++) {  \
```

Comment:
The underscore prefixes on `__i` clash with the existing conventions and don't really add anything. Should just be `i`.

Reply:
actually 'i' is really bad name as it's commonly used for counter and it may clash. let's call it 'i_'

Reply:
@flowher Actually `i_` might be a good idea here as you mentioned.

Otherwise, we're going to have some subtle macro bugs like:
```c
int i = 3;
const char *ret = NULL;
ARRAY_SEARCH_BY_ID(hash_alg_map, type, name, i, ret);
// ret is still NULL here if variable name is i
```

Reply:
yes, this macro was OK initially